iib/workers/tasks/build_create_empty_index.py

```diff
@@ -0,0 +1,152 @@
+    if not operators:
+        return []
+
+    # Transform returned data to parsable json
+    present_operators = []
+    new_operators = [json.loads(operator) for operator in re.split(r'(?<=})\n(?={)', operators)]
+
+    for operator in new_operators:
+        present_operators.append(operator['name'])
+
+    return present_operators
+
+
+@app.task
+@request_logger
+def handle_create_empty_index_request(
+    from_index, request_id, binary_image=None, labels=None, binary_image_config=None
+):
+    """Coordinate the the work needed to create the index image with labels.
+
+    :param str from_index: the pull specification of the container image containing the index that
+        the index image build will be based from.
+    :param int request_id: the ID of the IIB build request
+    :param str binary_image: the pull specification of the container image where the opm binary
+        gets copied from.
+    :param dict labels: the dict of labels required to be added to a new index image
+    :param dict binary_image_config: the dict of config required to identify the appropriate
+        ``binary_image`` to use.
+    """
+    _cleanup()
+
+    dist_scope = labels.get('com.redhat.index.delivery.distribution_scope', None)
+
+    prebuild_info = prepare_request_for_build(
+        request_id,
+        RequestConfigCreateIndexImage(
+            _binary_image=binary_image,
+            from_index=from_index,
+            distribution_scope=dist_scope,
+            binary_image_config=binary_image_config,
+        ),
+    )
+    from_index_resolved = prebuild_info['from_index_resolved']
+
+    prebuild_info['labels'] = labels
+
+    _update_index_image_build_state(request_id, prebuild_info)
+    with tempfile.TemporaryDirectory(prefix='iib-') as temp_dir:
+        msg = 'Checking operators present in index image'
+        set_request_state(request_id, 'in_progress', msg)
+
+        operators = _get_present_operators(from_index_resolved, temp_dir)
+
+        set_request_state(request_id, 'in_progress', 'Removing operators from index image')
+
+        _opm_index_rm(temp_dir, operators, prebuild_info['binary_image'], from_index_resolved)
+
+        set_request_state(
+            request_id, 'in_progress', 'Getting and updating labels for new index image'
+        )
+
+        iib_labels = {
+            'com.redhat.index.delivery.version': prebuild_info['ocp_version'],
+            'com.redhat.index.delivery.distribution_scope': prebuild_info['distribution_scope'],
+        }
+
+        iib_labels.update(labels)
+        for index_label, value in iib_labels.items():
+            _add_label_to_index(index_label, value, temp_dir, 'index.Dockerfile')
+
+        arches = prebuild_info['arches']
+
+        for arch in sorted(arches):
+            _build_image(temp_dir, 'index.Dockerfile', request_id, arch)
+            _push_image(request_id, arch)
+
+        # If the container-tool podman is used in the opm commands above, opm will create temporary
```

Comment:
I don't think we are using `container-tool=podman` in the opm commands above. We can probably get rid of this.

Reply:
removed. you are right I check the opm command